microprot/scripts/pdb_search.py

```diff
@@ -0,0 +1,393 @@
+
+        # P-value The P-value is just the E-value divided by the number of
+        # sequences in the database. It is the probability that in a PAIRWISE
+        # comparison a wrong hit will score at least this good.
+        hit[_HEADER[-7]] = float(fields[-7])
+
+        # Raw score, does not include the secondary structure score
+        hit[_HEADER[-6]] = float(fields[-6])
+
+        # Secondary structure score This score tells how well the PSIPRED-
+        # predicted (3-state) or actual DSSP-determined (8-state) secondary
+        # structure sequences agree with each other. PSIPRED confidence values
+        # are used in the scoring, low confidences getting less statistical
+        # weight.
+        hit[_HEADER[-5]] = float(fields[-5])
+
+        # The number of aligned Match columns in the HMM-HMM alignment.
+        hit[_HEADER[-4]] = int(fields[-4])
+
+        # Range of aligned match states from query HMM
+        hit[_HEADER[-3]] = fields[-3]
+
+        # Range of aligned match states from target HMM
+        hit[_HEADER[-2]] = fields[-2]
+
+        # Number of match states in target HMM
+        hit[_HEADER[-1]] = int(fields[-1][1:-1])
+
+        return hit
+    except ValueError:
+        raise ValueError("Unexpected field. Check if line is a HHsearch hit"
+                         " summary line.")
+
+
+def _parse_hit_block(block):
+    """ Parse one alignment block of HHsearch output.
+
+    Unfortunately, there is no strict format description. It is unclear which
+    lines might appear. Identifier might have white spaces. I assume that there
+    will always be a line starting with 'Q Consensus', having some integer to
+    describe the start position of the query, followed by a part of the aligned
+    query sequence. Delimiter are unclear, I guess it is \s+.
+    Due to all those uncertainties the code might look confusing at first
+    sight.
+
+    Parameters
+    ----------
+    block : str
+        The lines of the HHsearch output regarding to one alignment of query
+        sequence and target HMM. It might be wrapped into several parts.
+
+    Returns
+    -------
+    A dict holding all information about the aligment.
+    """
+
+    hit = {}
+    lines = block.split('\n')
+
+    # Index of hit
+    hit['No'] = int(lines[0].split()[1])
+
+    # description of domain
+    hit['Hit'] = lines[1][1:]
+
+    # line with stats
+    for field in lines[2].rstrip().split('  '):
+        key, value = field.split('=')
+        hit[key] = value
+    hit['Similarity'] = float(hit['Similarity'])
+    hit['Score'] = float(hit['Score'])
+    hit['Probab'] = float(hit['Probab'])
+    hit['Identities'] = float(hit['Identities'][:-1])/100
+    hit['Aligned_cols'] = int(hit['Aligned_cols'])
+    hit['E-value'] = float(hit['E-value'])
+    hit['Sum_probs'] = float(hit['Sum_probs'])
+    hit['Template_Neff'] = float(hit['Template_Neff'])
+
+    idx = 4
+    block = {}
+    while(idx+1 < len(lines)):
```

Comment:
This is such a complicated and elegant string processing function! Looks good to me.